test/e2e/page-objects/pages/snap-list-page.ts

```diff
@@ -0,0 +1,76 @@
+import { Driver } from '../../webdriver/driver';
+
+class SnapListPage {
+  private readonly driver: Driver;
+
+  private readonly closeModalButton = 'button[aria-label="Close"]';
+
+  private readonly continueRemoveSnapButton = {
+    tag: 'button',
+    text: 'Continue',
+  };
+
+  private readonly continueRemoveSnapModalMessage = {
+    tag: 'p',
+    text: 'Removing this Snap removes these accounts from MetaMask',
+  };
+
+  private readonly noSnapInstalledMessage = {
+    tag: 'p',
+    text: "You don't have any snaps installed.",
+  };
+
+  private readonly removeSnapButton = '[data-testid="remove-snap-button"]';
+
+  private readonly removeSnapConfirmationInput =
+    '[data-testid="remove-snap-confirmation-input"]';
+
+  private readonly removeSnapConfirmButton = {
+    tag: 'button',
+    text: 'Remove Snap',
+  };
+
+  private readonly snapListItem = '.snap-list-item';
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+  }
+
+  /**
+   * Removes a snap by its name from the snap list.
+   *
+   * @param snapName - The name of the snap to be removed.
+   */
+  async removeSnapByName(snapName: string): Promise<void> {
+    console.log('Removing snap on snap list page with name: ', snapName);
+    await this.driver.clickElement({ text: snapName, css: this.snapListItem });
+
+    const removeButton = await this.driver.findElement(this.removeSnapButton);
+    await this.driver.scrollToElement(removeButton);
+    await this.driver.clickElement(this.removeSnapButton);
```

Comment:
:information_source:  the fact that we need to do this scrolling to element previous to clicking it, is due to a bug into the Snap test dapp page.
I added a fix into the Snap test dapp page [here](https://github.com/MetaMask/snaps/pull/2782)  which should mitigate the flaky issue of scrolling as well as clicking to elements in the snap test dapp.

Once the snaps team releases the new version with the fix, we'll be able to remove this scrolling steps and just work normally with the clickElement (which already does the scrolling).

See more context in the PR above (as well as all the issues related to that)

Reply:
Thanks for the information ! I have added the necessary TODO comment for it